helpers/floats.go

```diff
@@ -0,0 +1,32 @@
+package helpers
+
+import "math"
+
+const (
+	DefaultEpsilon = 0.000000001
+)
+
+var (
+	FloatMinNormal = math.Float64frombits(0x00800000)
+)
+
+func NearlyEqual(a, b float64) bool {
+	return NearlyEqualEpsilon(a, b, DefaultEpsilon)
+}
+
+// NearlyEqual float points covering some edge cases as per https://floating-point-gui.de/errors/comparison/#look-out-for-edge-cases
+func NearlyEqualEpsilon(a, b float64, epsilon float64) bool {
+	absA := math.Abs(a)
+	absB := math.Abs(b)
+	diff := math.Abs(a - b)
+
+	if a == b { // shortcut, handles infinities
+		return true
+	} else if a == 0 || b == 0 || diff < FloatMinNormal {
```

Comment:
`diff` shall be `absA + absB`

Reply:
hmm interesting, actually converted the code the shows up in duckduckgo which is from stackoverflow (but wanted to link to original author), comparing what's on the site and what shows there is different, must have been updated
![image](https://user-images.githubusercontent.com/16015083/117423167-b545d900-af20-11eb-894a-a18887344f93.png)

Reply:
good find. will update.